shared/modules/Storage/ReadOnlyNetworkStore.ts

```diff
@@ -0,0 +1,115 @@
+import log from 'loglevel';
+import { isErrorWithMessage } from '@metamask/utils';
+import getFetchWithTimeout from '../fetch-with-timeout';
+import type Migrator from '../../../app/scripts/lib/migrator';
+import { type IntermediaryStateType, BaseStorage } from './Storage';
+
+const fetchWithTimeout = getFetchWithTimeout();
+
+const FIXTURE_SERVER_HOST = 'localhost';
+const FIXTURE_SERVER_PORT = 12345;
+const FIXTURE_SERVER_URL = `http://${FIXTURE_SERVER_HOST}:${FIXTURE_SERVER_PORT}/state.json`;
+
+/**
+ * A read-only network-based storage wrapper
+ */
+export default class ReadOnlyNetworkStore extends BaseStorage {
+  #initialized: boolean;
```

Comment:
QQ: why are we using `#` instead of `private` keyword here ?

Reply:
Given that background.js is still just javascript, the private javascript field ensures that these properties will not be misused or unintentionally modified.